src/OSInfo.cpp

```diff
@@ -23,6 +23,21 @@
 #include "Version.h"
 
 #if defined(Q_OS_WIN)
+// regString converts a wchar_t string of size to a
+// QString. If the string contains a NUL value, that
+// NUL value will terminate the string.
+static QString regString(wchar_t *string, int size) {
+	// If string contains a NUL, set the size such that
+	// the NUL is not included in the returned string.
+	for (int i = 0; i < size; i++) {
+		if (string[i] == 0) {
+			size = i;
+			break;
+		}
+	}
+	return QString::fromWCharArray(string, size);
```

Comment:
There are already functions for what the loop does in the ```*nlen``` family:
```cpp
static QString regString(wchar_t *string, int size) {
    return QString::fromWCharArray(string, wcsnlen(string, size));
}
```
Ought to work the same I think (not tested).